### PR TITLE
Blazor Hybrid WinForms tutorial (PR3)

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #24956

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/hybrid/tutorials/windows-forms?view=aspnetcore-6.0&branch=pr-en-us-24989)

Naming ...

* The final app naming is "Windows Forms" **_before_** "Blazor" to match what we have in the .NET MAUI case. ".NET MAUI Blazor app" ... so ... "Windows Forms Blazor app" is the way that I went. Obviously (and given our earlier discussion on official names of Blazor Hybrid apps still a bit of a *WIP*), I'm just trying to maintain consistency.
* WRT "WinForms" vs "Windows Forms," I checked the Windows Forms doc set to see what they do. They **_highly prefer_** the formal name "Windows Forms." They don't totally avoid "WinForms." They use it rarely ... e.g., in some sample app scenarios. I took the same approach here. "WinForms" is used in sample naming but not in the topic, which consistently goes with "Windows Forms."

### CODE REVIEW

**Review by danroth27 — CHANGES_REQUESTED** [review]
Thanks for putting this together, @guardrex. 

I would prefer to simplify this tutorial to setup a single Razor component instead of the entire default Blazor project template. That should allow us to give inline instructions instead of copying from other projects.

Here are the high-level steps that I'd like to use:

1. Create a new Windows Forms app
2. Install the [Microsoft.AspNetCore.Components.WebView.WindowsForms](https://nuget.org/packages/Microsoft.AspNetCore.Components.WebView.WindowsForms) using the NuGet package manager
3. Update the .csproj file to change the SDK to `Microsoft.NET.Sdk.Razor`
4. Add an *_Imports.razor* file with `@using Microsoft.AspNetCore.Components.Web`
5. Add a *wwwroot* folder
6. Add *wwwroot\index.html* with the following:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>WinFormsApp1</title>
    <base href="/" />
    <link href="css/app.css" rel="stylesheet" />
    <link href="WinFormsApp1.styles.css" rel="stylesheet" />
</head>

<body>

    <div id="app">Loading...</div>

    <div id="blazor-error-ui">
        An unhandled error has occurred.
        <a href="" class="reload">Reload</a>
        <a class="dismiss">🗙</a>
    </div>

    <script src="_framework/blazor.webview.js"></script>

</body>

</html>
```

7. Add *wwwroot/css/app.css* with the following

```css
html, body {
    font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
}

.btn-primary {
    color: #fff;
    background-color: #1b6ec2;
    border-color: #1861ac;
}

.valid.modified:not([type=checkbox]) {
    outline: 1px solid #26b050;
}

.invalid {
    outline: 1px solid red;
}

.validation-message {
    color: red;
}

#blazor-error-ui {
    background: lightyellow;
    bottom: 0;
    box-shadow: 0 -1px 2px rgba(0, 0, 0, 0.2);
    display: none;
    left: 0;
    padding: 0.6rem 1.25rem 0.7rem 1.25rem;
    position: fixed;
    width: 100%;
    z-index: 1000;
}

    #blazor-error-ui .dismiss {
        cursor: pointer;
        position: absolute;
        right: 0.75rem;
        top: 0.5rem;
    }
```

8. Update the .csproj file to copy `wwwroot\**` to the output directory:

> I'm following up on how we can get rid of this step but for now I think it's needed.

```xml
<ItemGroup>
  <Content Update="wwwroot\**" CopyToOutputDirectory="PreserveNewest" />
</ItemGroup>
```

9. Add a *Counter.razor* file with the default `Counter` component code.

```razor
<h1>Counter</h1>

<p>Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>

@code {
    private int currentCount = 0;

    private void IncrementCount()
    {
        currentCount++;
    }
}
```

10. Double-click on `Form1.cs` to open the designer.
11. Use View > Toolbox, and find `BlazorWebView`
12. Drag `BlazorWebView` into the `Form1` designer.
13. Right-click on this new item and choose *Properties*. Change its *Dock* value to *Fill* by clicking on the central rectangle:
14. Inside the `Form1` constructor, underneath `InitializeComponent()`, add the following code:

```cs
var services = new ServiceCollection();
services.AddBlazorWebView();
blazorWebView1.HostPage = "wwwroot\\index.html";
blazorWebView1.Services = services.BuildServiceProvider();
blazorWebView1.RootComponents.Add<Counter>("#app");
```

15. Run the app.

File: aspnetcore/blazor/hybrid/tutorials/windows-forms.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            };
+            blazor.RootComponents.Add<Main>("#app");
+            Controls.Add(blazor);
+        }
+
+    }
+}
+```
+
+In the `Program.cs` file, replace the code in the `Main` method with the following:
```
Comment (danroth27) [inline aspnetcore/blazor/hybrid/tutorials/windows-forms.md:240]: I don't this this step is necessary, so let's drop it.

File: aspnetcore/blazor/hybrid/tutorials/windows-forms.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    </Found>
+    <NotFound>
+        <LayoutView Layout="@typeof(MainLayout)">
+            <p>Sorry, there's nothing at this address.</p>
+        </LayoutView>
+    </NotFound>
+</Router>
+```
+
+Open the `Form1.Designer.cs` file from **Solution Explorer**:
```
Comment (danroth27) [inline aspnetcore/blazor/hybrid/tutorials/windows-forms.md:147]: I don't think any modifications of Form1.Designer.cs are necessary, so let's also drop this section.

**Comment by guardrex** [comment]
> I would prefer to simplify this tutorial to setup a single Razor component instead of the entire default Blazor project template. That should allow us to give inline instructions instead of copying from other projects.

Sure thing. :+1:

Due to the need to rewrite this, I probably won't have the WPF tutorial up until Wednesday morning. When I reach it, I'll write the WPF tutorial in the same vein as the rewrite guidance.

Just to clarify to my colleagues here 👀 ... because it kind'a looks like I made a mistake or **_went rogue_** 🏴‍☠️😄: I was going off of the [project sample provided](https://github.com/danroth27/BlazorWinForms). The sample is a full Blazor project template-based WinForms app ... actually, it's a little worse than that ... it's really a **_blend of both hosting models_** into a WinForms app ... so a blend of **_three apps_**. We'll take this in a different direction now.

**Comment by guardrex** [comment]
❓s on the updates ...

1. Instead of `using` statements in `Form1.cs`, do you prefer that we include them in `_Imports.razor`?
2. Do you want to include any further namespaces in the `_Imports.razor` file?
3. I suppose ur good on naming the project `WinFormsBlazor`, which is what I did on the PR ... seems nicer to me than `WinFormsApp1`. btw- If we keep this naming here, I'll update the MAUI tutorial later to use `MauiBlazor` (or similar naming). It's currently using kind'a *crappy-er-ish* 😄 `MauiApp1` naming.
4. Leave the stylesheet link for `WinFormsApp1.styles.css` in `index.html`?
5. Do you want to send me a rounded-corners image for `running-app.png`? If so, please snap it with a little extra space around the image on a white or black background to facilitate the Photoshopping step here.

**Comment by guardrex** [comment]
I'll be on stand-by for feedback and updates.

I'm around the 🏠 all day right into the evening ... **_WELL_** ... until the [US Curling Teams](https://www.usacurling.org/national-team) get on the ice!

# *Go Team Tabitha!*

tabitha

# *Go Team Shuster!*

shuster

File: aspnetcore/blazor/hybrid/tutorials/windows-forms.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+:::image type="content" source="windows-forms/_static/create-project.png" alt-text="Create a new project in Visual Studio.":::
+
+In the **Configure your new project** dialog, set the **Project name** to **`WinFormsBlazor`**, choose a suitable location for the project, and select the **Next** button. Using `WinFormsBlazor` as the project name matches the donor Blazor project names created in the preceding section, which aligns the namespaces of the three projects.
+
+:::image type="content" source="windows-forms/_static/configure-project.png" alt-text="Configure the project.":::
+
+In the **Additional information** dialog, select the framework version, which must be .NET 6.0 or later. Select the **Create** button.
+
+Use [NuGet Package Manager](/nuget/consume-packages/install-use-packages-visual-studio) to install the [`Microsoft.AspNetCore.Components.WebView.WindowsForms`](https://nuget.org/packages/Microsoft.AspNetCore.Components.WebView.WindowsForms) NuGet package.
```
Comment (danroth27) [inline aspnetcore/blazor/hybrid/tutorials/windows-forms.md:45]:
```suggestion
Use [NuGet Package Manager](/nuget/consume-packages/install-use-packages-visual-studio) to install the [`Microsoft.AspNetCore.Components.WebView.WindowsForms`](https://nuget.org/packages/Microsoft.AspNetCore.Components.WebView.WindowsForms) preview NuGet package.
```

**Pull request merged**